pkg/lib/cockpit-components-truncate.jsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2024 Mohd Mohsin
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* This is our version of the PatternFly Truncate component. We have
+   it since we don't want Patternfly's unconditional tooltip.
+
+   See https://github.com/patternfly/patternfly/issues/6253
+ */
+
+import * as React from "react";
+
+const minWidthCharacters = 16;
+
+const sliceContent = (str, slice) => [
+    str.slice(0, str.length - slice),
+    str.slice(-slice),
+];
+
+export const Truncate = ({
+    className,
+    position = "end",
+    trailingNumChars = 7,
+    content,
+    ...props
+}) => {
+    return (
+        <div>
+            <span className="pf-v5-c-truncate" {...props}>
+                {position === "end" && (
+                    <span className="pf-v5-c-truncate__start">
+                        {content}
+                    </span>
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19837-20240129-120931-758dd2ff-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)